In [15]:
import numpy as np
import seaborn as sns
import control as cl
import scipy as sp
import pandas as pd

## Explorando estrutura de 'classes'

In [16]:
# carregar o arquivo classes.mat
data1 = sp.io.loadmat('classes.mat')

In [17]:
sp.io.whosmat('mat_classes.mat')

[('classes', (1000, 2561), 'double')]

In [27]:

# carrega o arquivo .mat
mat_data1 = sp.io.loadmat('mat_classes.mat')

# extrai a matriz "classes" do dicionário mat_data
classes = mat_data1['classes']

# imprime o número de linhas e colunas da matriz
print(classes.shape)

# imprime o primeiro elemento da matriz
print(classes[0,0])

# imprime o nome da primeira coluna da matriz (se houver)
print(classes.dtype.name[1])

(1000, 2561)
-0.0488281249677312
l


> ou seja, é a definição de uma espécie de tabela 1000 linhas x 2561 colunas. Infelizmente não há informações mais detalhadas, as colunas não tem informação.

In [28]:
# extrair a variável 'classes' do dicionário data
classes = data1['classes']

# criar um dataframe pandas com a matriz numpy 'classes'
df = pd.DataFrame(classes)

# imprimir o dataframe
print(df)

         0         1         2         3         4         5         6     \
0   -0.048828  0.216675  0.234985  0.390625  0.003052 -0.265503 -0.143433   
1   -0.137329  0.036621  0.231934  0.466919  0.161743 -0.112915 -0.277710   
2    0.146484  0.030518 -0.103760  0.119019 -0.250244 -0.201416 -0.170898   
3   -0.094604 -0.048828 -0.085449 -0.192261 -0.094604 -0.067139 -0.045776   
4    0.265503  0.399780  0.073242  0.299072 -0.198364 -0.119019 -0.064087   
..        ...       ...       ...       ...       ...       ...       ...   
995 -0.054932  0.027466  0.097656  0.186157  0.204468  0.170898  0.033569   
996 -0.180054 -0.265503 -0.219727 -0.079346  0.143433  0.280762  0.262451   
997  0.372314  0.149536 -0.149536 -0.418091 -0.469971 -0.308228 -0.061035   
998 -0.088501 -0.115967 -0.180054 -0.122070  0.012207  0.207520  0.390625   
999 -0.039673  0.009155  0.024414  0.073242  0.067139  0.164795  0.357056   

         7         8         9     ...      2551      2552      2553  \
0  

## Explorando estrutura de 'vibrations'

In [32]:
# carregar o arquivo classes.mat
data2 = sp.io.loadmat('mat_vibration.mat')

In [34]:
# carrega o arquivo .mat
mat_data2 = sp.io.loadmat('mat_vibration.mat')

# extrai a matriz "classes" do dicionário mat_data
mat_vibration = mat_data2['A']

# imprime o número de linhas e colunas da matriz
print(classes.shape)

# imprime o primeiro elemento da matriz
print(classes[0,0])

# imprime o nome da primeira coluna da matriz (se houver)
print(classes.dtype.name[1])

(1000, 2561)
-0.0488281249677312
l


In [30]:
sp.io.whosmat('mat_vibration.mat')

[('A', (3, 3), 'double'),
 ('B', (2, 3), 'double'),
 ('C', (5, 3), 'double'),
 ('D', (3, 2), 'double'),
 ('E', (3, 5), 'double'),
 ('N15_M07_F10_K001_1', (1, 1), 'struct'),
 ('N15_M07_F10_KA04_1', (1, 1), 'struct'),
 ('N15_M07_F10_KA15_1', (1, 1), 'struct'),
 ('N15_M07_F10_KB23_1', (1, 1), 'struct'),
 ('N15_M07_F10_KB27_1', (1, 1), 'struct'),
 ('N15_M07_F10_KI04_1', (1, 1), 'struct'),
 ('ans', (1, 1), 'struct'),
 ('class01', (100, 2561), 'double'),
 ('class02', (100, 2561), 'double'),
 ('class03', (100, 2561), 'double'),
 ('class04', (100, 2561), 'double'),
 ('class05', (100, 2561), 'double'),
 ('class06', (100, 2561), 'double'),
 ('class1', (100, 2560), 'double'),
 ('class2', (100, 2560), 'double'),
 ('class3', (100, 2560), 'double'),
 ('class4', (100, 2560), 'double'),
 ('class5', (100, 2560), 'double'),
 ('class6', (100, 2560), 'double'),
 ('mat_vibration', (600, 2561), 'double'),
 ('matrix', (100, 2560), 'double'),
 ('n_var', (1, 1), 'double'),
 ('phase_current_1_K001_1', (1, 25600

In [13]:
# imprimir as dimensões da matriz 'classes'
print(classes.shape)

(1000, 2561)


In [12]:
# extrair a variável 'classes' do dicionário data
mat_vibrations = data2['classes']

# criar um dataframe pandas com a matriz numpy 'classes'
df = pd.DataFrame(classes)

# imprimir o dataframe
print(df)

         0         1         2         3         4         5         6     \
0   -0.048828  0.216675  0.234985  0.390625  0.003052 -0.265503 -0.143433   
1   -0.137329  0.036621  0.231934  0.466919  0.161743 -0.112915 -0.277710   
2    0.146484  0.030518 -0.103760  0.119019 -0.250244 -0.201416 -0.170898   
3   -0.094604 -0.048828 -0.085449 -0.192261 -0.094604 -0.067139 -0.045776   
4    0.265503  0.399780  0.073242  0.299072 -0.198364 -0.119019 -0.064087   
..        ...       ...       ...       ...       ...       ...       ...   
995 -0.054932  0.027466  0.097656  0.186157  0.204468  0.170898  0.033569   
996 -0.180054 -0.265503 -0.219727 -0.079346  0.143433  0.280762  0.262451   
997  0.372314  0.149536 -0.149536 -0.418091 -0.469971 -0.308228 -0.061035   
998 -0.088501 -0.115967 -0.180054 -0.122070  0.012207  0.207520  0.390625   
999 -0.039673  0.009155  0.024414  0.073242  0.067139  0.164795  0.357056   

         7         8         9     ...      2551      2552      2553  \
0  